ReactiveCocoa/Swift/SignalProducer.swift

```diff
@@ -1695,26 +1682,27 @@ private final class DeallocationToken {
 	}
 }
 
-extension SignalProducer {
-	private static func bufferingProducer(upTo capacity: Int) -> (SignalProducer, Signal<Value, Error>.Observer) {
-		precondition(capacity >= 0, "Invalid capacity: \(capacity)")
+private struct ReplayState<Value, Error: ErrorProtocol> {
+	let producer: SignalProducer<Value, Error>
+	let observer: SignalProducer<Value, Error>.ProducedSignal.Observer
```

Comment:
This seems a bit odd to me for a few reasons:

1. `ReplayState` is used with `Atomic` by `replayLazily` but itself uses an `Atomic<BufferState<Value, Error>>`.
2. `BufferState` is only used inside `ReplayState`.
3. `ReplayState` has a producer and observer, which means that it's more than just state information

How would you feel about attempting to unify this code a bit more?